infra/cifuzz/cifuzz.py

```diff
@@ -0,0 +1,149 @@
+      project_name, repo_name=project_repo_name, src_dir=src)
+  if not inferred_url or not oss_fuzz_repo_name:
+    logging.error('Could not detect repo from project %s.', project_name)
+    return False
+
+  # Checkout projects repo in the shared volume.
+  build_repo_manager = repo_manager.RepoManager(inferred_url,
+                                                git_workspace,
+                                                repo_name=oss_fuzz_repo_name)
+  try:
+    build_repo_manager.checkout_commit(commit_sha)
+  except repo_manager.RepoManagerError:
+    logging.error('Error: Specified commit does not exist.')
+    # NOTE: remove return statement for testing.
+    return False
+
+  command = [
+      '--cap-add', 'SYS_PTRACE', '-e', 'FUZZING_ENGINE=libfuzzer', '-e',
+      'SANITIZER=address', '-e', 'ARCHITECTURE=x86_64'
+  ]
+  container = utils.get_container()
+  if container:
+    command += ['-e', 'OUT=' + out_dir, '--volumes-from', container]
+    bash_command = 'rm -rf {0} && cp -r {1} {2} && compile'.format(
+        os.path.join(src, oss_fuzz_repo_name, '*'),
+        os.path.join(git_workspace, oss_fuzz_repo_name), src)
+  else:
+    command += [
+        '-e', 'OUT=' + '/out', '-v',
+        '%s:%s' % (os.path.join(git_workspace, oss_fuzz_repo_name),
+                   os.path.join(src, oss_fuzz_repo_name)), '-v',
+        '%s:%s' % (out_dir, '/out')
+    ]
+    bash_command = 'compile'
+
+  command.extend([
+      'gcr.io/oss-fuzz/%s' % project_name,
+      '/bin/bash',
+      '-c',
+  ])
+  command.append(bash_command)
+
+  if helper.docker_run(command):
+    logging.error('Error: Building fuzzers failed.')
+    return False
+  return True
+
+
+def run_fuzzers(project_name, fuzz_seconds, out_dir):
+  """Runs a all fuzzers for a specific OSS-Fuzz project.
+
+  Args:
+    project_name: The name of the OSS-Fuzz project being built.
+    fuzz_seconds: The total time allotted for fuzzing.
+    out_dir: The location in the shared volume to store output artifacts.
+
+  Returns:
+    (True if run was successful, True if bug was found False if not).
+  """
+  if not out_dir or not os.path.exists(out_dir):
+    logging.error('Error: Unreachable out_dir argument.')
+    return False, False
+
+  if not fuzz_seconds or fuzz_seconds < 1:
+    logging.error('Error: fuzz_seconds argument must be greater than 1.')
+    return False, False
+
+  fuzzer_paths = utils.get_fuzz_targets(out_dir)
+  if not fuzzer_paths:
+    logging.error('Error: No fuzzers were found in out directory.')
+    return False, False
+
+  fuzzer_timeout = fuzz_seconds // len(fuzzer_paths)
+
+  for fuzzer_path in fuzzer_paths:
+    target = fuzz_target.FuzzTarget(project_name, fuzzer_path, fuzzer_timeout)
+    test_case, stack_trace = target.fuzz()
+    if not test_case or not stack_trace:
+      logging.debug('Fuzzer %s, finished running.', target.target_name)
+    else:
+      logging.debug("Fuzzer %s, Detected Error: %s", target.target_name,
```

Comment:
nit: Drop capital D in Detected and end message with period.

Reply:
Done.